### ISSUE TITLE
Unable to authenticate in test setup with pam

### ISSUE THREAD

**Issue by harkx** [tinyclues/saltpad#51]
I followed the README and created a virtualenv for my test setup. The settings are:
```python
SECRET_KEY = 'xxxx'
HOST = "0.0.0.0"
EAUTH = "pam"
```
System is a Debian Wheezy. Saltpad is running on my master. Salt(-api) (2014.7.5) is configured to use pam, this part works when I use curl to authenticate and run commands.

When trying to login via saltpad (I get the login screen), I enter my user+pass and get returned to the login screen. (no error, it does display **Hi MyName**) 

The only error I can find is after enabling debug in the salt master:
`2015-04-27 13:05:55,147 [salt.master][WARNING ] Authentication failure of type "token" occurred for user MyName.`

I suppose I'm doing something wrong, can anyone lead me into the right direction?

**Comment by bbinet** [comment]
what is you `external_auth` configuration?
it should look like:
```
external_auth:
  pam:
    my_user:
      - .*
      - '@runner'
      - '@wheel'
```

**Comment by harkx** [comment]
Hi, yes, correct, this is in /etc/salt/master:

```
external_auth:
  pam:
    kristof:
      - .*
```

**Comment by bbinet** [comment]
So I think you should also add the following permissions
```
      - '@runner'
      - '@wheel'
```

**Comment by harkx** [comment]
Oh boy, that's it.. Sorry! I didn't read that correctly. (didn't know that was mandatory) .. Sorry, but thanks a lot for your help!!

**Comment by Lothiraldan** [comment]
Thanks @bbinet for the debugging and @harkx for the issue, I will make sure it's well documented in the README and add a better error message.

**Comment by ace-han** [comment]
Hello, there

I would request a walkthrough guide on pam as I am new to salt and pam.
I got stuck for all night fitghting with pam  setting and got no luck.

It's pretty funny that I could not get a good doc about how to setup an valid account via pam to try on saltpad...

Please help me out...

**Comment by harkx** [comment]
Well @ace-han , I did it like this.. (follow the install guide to put saltpad on your system)

You create a linux user with your name and give it your preferred password:
```bash
adduser acehan
passwd acehan</code
```
After that, you literally put this in your /etc/salt/master file:

```bash
external_auth:
  pam:
    acehan:
      - .*
      - '@runner'
      - '@wheel'
```

That should do it. You can use your newly created user and password to login to saltpad.

**Comment by ace-han** [comment]
@harkx  Thanks, I think it is will be better to add these few lines to `READ.md`
It would be definitely a time saver.
I have set it up via `mysql`. And written a much friendly setup guide at 
[salt-master_n_salt-api_setup.md](https://github.com/ace-han/saltstack-mark/blob/master/salt-master_n_salt-api_setup.md)
[saltpad_setup.md](https://github.com/ace-han/saltstack-mark/blob/master/saltpad_setup.md)

**Comment by Lothiraldan** [comment]
@ace-han Well noted, I will add a note about it

**Comment by KingJ** [comment]
I'm also running in to this issue, but only when i'm using salt-api's CherryPi module via uwsgi and nginx. If I enable CherryPi's HTTP server in the master configuration and point SaltPad at that, I can log in to SaltPad without issue. I tried running the Salt Master with debug logging, and the only log lines generated during an unsuccessful login are;

```
2015-06-27 14:24:29,181 [salt.master      ][INFO    ][27345] Clear payload received with command mk_token
2015-06-27 14:24:29,208 [salt.utils.lazy  ][DEBUG   ][27345] LazyLoaded pam.auth
```

I get the same lines for a successful login via CherryPi, along with many more as SaltPad pulls in data over the API.

I'm inclined to believe this is a configuration issue on my end, but i'm out of debugging ideas. I've verified that I can log in to the API using a curl POST and that i've got the appropriate permissions for my user in the master configuration so i'm utterly perplexed by this! If anyone can offer any insight, i'd greatly appreciate it.
